### PR TITLE
Triage categorization design 

### PR DESCRIPTION
**Pull request by chcosta**
Fixes https://github.com/dotnet/core-eng/issues/8284

@mmitche, do we have other GitHub bots already?  I feel like the answer is no.  I suppose we should do some research to find out what the runtime bot does (pretty certain they have one) and if it's something we can / should use for this (though then there becomes an ownership question).

### CODE REVIEW

File: Documentation/Projects/PKPIs/Triage-Design.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+![Chart](./Triage-Design.png)
+
+### Sql table
+
+- Create a table that contains BuildId / RecordId, category, GitHub issue link
+
+### PowerBI
+
+- Link to Sql table, latest row in Sql table  for a given BuildId / RecordId supersede existing categorization
```
Comment (jcagme) [inline Documentation/Projects/PKPIs/Triage-Design.md]: >   [](start = 44, length = 2)

Two spaces

File: Documentation/Projects/PKPIs/Triage-Design.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+- Add column which contains links that will create GitHub issues if no GitHub issue is already present for a given BuildId / RecordId
```
Comment (jcagme) [inline Documentation/Projects/PKPIs/Triage-Design.md]: >Add column which contains links that will create GitHub issues if no GitHub issue is already present for a given BuildId / RecordId [](start = 2, length = 131)

What creates the GH issues? Clicking the link?

**Comment by mmitche** [comment]
> @mmitche, do we have other GitHub bots already? I feel like the answer is no. I suppose we should do some research to find out what the runtime bot does (pretty certain they have one) and if it's something we can / should use for this (though then there becomes an ownership question).

Yeah, though not ones we own.  There is Karel's bot, msft bot, office fabric bot.

File: Documentation/Projects/PKPIs/Triage-Design.md (hunk 2 above)
Comment (chcosta) [reply Documentation/Projects/PKPIs/Triage-Design.md]: Yes, ie, the URL itself takes you to the GitHub create issues page with prepopulated data, it's then up to you to actively click the "Submit new issue" button.

File: Documentation/Projects/PKPIs/Triage-Design.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+### GitHub bot
+
+- for created issues (if it's a triage issue)
+
+  - Determines if an issue already exists
+
+    - if yes, then add a link and close
+
+    - if no, then add an entry into the sql table with github link and category=Triage
```
Comment (Chrisboh) [inline Documentation/Projects/PKPIs/Triage-Design.md]: What is the process by which we are going to add the entry to sql?

File: Documentation/Projects/PKPIs/Triage-Design.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+[Telemetry categorization](../DevOps/CI/Telemetry-Guidance.md) has allowed us to bucket Azure DevOps failures.  Improvements to telemetry categorization (mostly) only affect future builds.  There is no method to fix categorization for builds that have already completed or which were improperly categorized.
+
+## Proposed solution
+
+![Chart](./Triage-Design.png)
+
+### Sql table
+
+- Create a table that contains BuildId / RecordId, category, GitHub issue link
```
Comment (Chrisboh) [inline Documentation/Projects/PKPIs/Triage-Design.md]: Do you have any concerns having the correction data in SQL and the build data in Kusto?  I ask for future one off reporting which is sometimes just done from Kusto.

File: Documentation/Projects/PKPIs/Triage-Design.md (hunk 3 above)
Comment (chcosta) [reply Documentation/Projects/PKPIs/Triage-Design.md]: the bot adds the entry

File: Documentation/Projects/PKPIs/Triage-Design.md (hunk 4 above)
Comment (chcosta) [reply Documentation/Projects/PKPIs/Triage-Design.md]: No, we already merge Kusto and sql data in the report.  So, from an implementation perspective, no, I don't.  From a cleanliness perspective (and simplicity), I would prefer we just had one source (Kusto, sql, or other), but that's tangential.

**Comment by chcosta** [comment]
The pitfall I see with this approach, I don't have a good solution for.  That issue is that there will likely be some timeframe when someone has created a GH issue and that isn't reflected in the PowerBi report which will mean that they could click the same link and create another GH issue or don't have a direct way to get from the PowerBi report to the previously reported issue.  We might be able to preserve the state in power bi and report a message instead of permitting the use of the GH creation link, but I'm not 100% clear on what Power BI will permit here and if it will persist to others.

**Comment by jonfortescue** [comment]
you should move the "Helix" and "PowerBI" labels down so they fit more neatly in the image.

Also, I realize that you have screenshots in this so it is likely not possible, but in general I prefer SVGs since they scale better, are committed as code, and are easier to edit.

**Comment by chcosta** [comment]
Got an ack from @shawnro and @mmitche on this plan.

**Comment by chcosta** [comment]
Got an ack from @mattgal on this issue.  I'm going to merge this as the P.O.R.  There will likely be some details to sort out (labels vs comments, etc...) but I'll move forward.

**Comment by chcosta** [comment]
Also, note, the SQL table mentioned in the doc, is actually one of our Kusto tables.

**Pull request merged**